app/src/features/surveys/view/survey-animals/TelemetryDeviceForm.tsx

```diff
@@ -156,7 +157,13 @@ const DeviceFormSection = ({ values, index, mode, removeAction }: IDeviceFormSec
         </Typography>
         <Grid container spacing={3}>
           <Grid item xs={6}>
-            <CustomTextField label="Device ID" name={`${index}.device_id`} other={{ disabled: mode === 'edit' }} />
+            <CustomNumberField
+              label="Device ID"
+              name={`${index}.device_id`}
+              disabled={mode === 'edit'}
+              min={1}
+              max={2147483647}
```

Comment:
nit: I think it could make sense to make this a constant somewhere, something like PG_MAX_INT. Makes it more clear that this max value was chosen for a particular reason.